src/logger/httpLoggerRules_test.go

```diff
@@ -286,195 +286,662 @@ func TestUsesRemoveRules(t *testing.T) {
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(mockHtml))
+	mockResponse.Request = request
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "/response_body/ replace /[0-9\\.\\-\\/]{9,}/, /xyxy/")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello xyxy!</html>\"],"), "custom string not replaced in queue")
+
+	request = helper.MockRequestWithJson2()
+	mockResponse = helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml))
+	mockResponse.Request = request
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !World!, !<b>$&</b>!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello <b>World</b>!</html>\"],"), "custom string not replaced in queue")
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !(World)!, !<b>$1</b>!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello <b>World</b>!</html>\"],"), "custom string not replaced in queue")	
+
+	request = helper.MockRequestWithJson2()
+	mockResponse = helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml5))
+	mockResponse.Request = request
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !<input([^>]*)>([^<]*)</input>!, !<input$1></input>!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>\\n<input type=\\\"hidden\\\"></input>\\n<input class='foo' type=\\\"hidden\\\"></input>\\n</html>\"],"), "custom string not replaced in queue")
+}
+
 // test uses sample rules
 
+func TestUsesSampleRules(t *testing.T) {
+	helper := GetTestHelper()
+
+	request := helper.MockRequestWithJson2()
+	mockResponse := helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml))
+	mockResponse.Request = request
+	
+	queue := make([]string, 0)
+
+	logger, err := NewHttpLoggerQueueRules(queue, "sample 10\nsample 99")
```

Comment:
So this assumes that our httplogger constructor will return and error... which is fine but we need to make a decision whether we want to do this.

I think we probably should for constructors that require further operations which might fail/return an error (i.e. passing rules requires parsing the rule to return the correct set of httprules).